youtube/feed.go

```diff
@@ -277,37 +278,45 @@ var (
 )
 
 func (p *Plugin) parseYtUrl(url string) (t ytUrlType, id string, err error) {
+	validateCaptures := func(expr *regexp.Regexp, i int) (id string, valid bool) {
```

Comment:
move this to Plugin struct instead of keeping as a lambda function

`func (p *Plugin) getIDFromRegexURL(expr *regexp.Regex, url string, index int) (id string, valid bool)`